pkg/model/diamond.go

```diff
@@ -0,0 +1,265 @@
+package model
+
+import (
+	"fmt"
+	"path"
+	"time"
+
+	"github.com/segmentio/ksuid"
+)
+
+// ConflictMode indicates the conflict detection mode defined for a diamond
+type ConflictMode string
+
+const (
+	// IgnoreConflicts is the diamond mode with which conflicts are not handled (latest win, no track kept of clobbered files)
+	IgnoreConflicts ConflictMode = "ignored"
+
+	// EnableCheckpoints is the diamond mode with which conflicts are explicitly handled and saved as "checkpoints" (incremental upload)
+	EnableCheckpoints ConflictMode = "enable-checkpoints"
+
+	// EnableConflicts is the diamond mode with which conflicts are detected and saved as "conflicts"
+	EnableConflicts ConflictMode = "enable-conflicts"
+
+	// ForbidConflicts is the diamond mode with which conflicts result in failure to commit
+	ForbidConflicts ConflictMode = "forbids-conflicts"
+)
+
+// DiamondState models the running status of an ungoing diamond workflow
+type DiamondState string
+
+const (
+	// DiamondInitialized is the state of an initialized diamond
+	DiamondInitialized DiamondState = "initialized"
+
+	// DiamondCommitting indicates that a commit is going on for this diamond
+	DiamondCommitting DiamondState = "committing"
+
+	// DiamondCancelling indicates that a cancel is going on for this diamond
+	DiamondCancelling DiamondState = "cancelling"
+
+	// DiamondRunning indicates that the diamond has started some splits
+	DiamondRunning DiamondState = "running"
+
+	// DiamondDone indicates the diamond has completed with a commit
+	DiamondDone DiamondState = "done"
+
+	// DiamondCancelled indicates the diamond has completed with a cancel
+	DiamondCancelled DiamondState = "cancelled"
+)
+
+const (
+	diamondDescriptorFile = "diamond.yaml"
```

Comment:
nit: to `const` expression with `bundleDescriptorFile`, etc. in `paths.go`

Reply:
the whole thing should move to a subpackage called something like `paths` or `metapaths`